### PR TITLE
optionally add html component to image extract

### PR DESCRIPTION
**Pull request by bjsi**
Optionally add html component to PDF extracts containing only images as requested by @nikedinikedi .    The option can be set in the PDF config settings. 

The only thing that seems off to me is that the titles of the extracts looks like the picture below. Let me know if this is an issue, I am more than willing to fix this, because admittedly this is a very simple solution.

image

### CODE REVIEW

**Comment by bjsi** [comment]
I struggled to add a better title, but this is what I came up with:

image

File: PDF/Viewer/IPDFViewer.SuperMemo.cs [hunk 1]
```diff
@@ -142,8 +144,14 @@ protected bool CreateSMExtract()
 
         contents.Add(new TextContent(true, text));
       }
-      else if (Config.ImageExtractAddHtml)
+      else if (Config.ImageExtractAddHtml && imgExtracts.Count > 0)
       {
+        extractTitle = $"image extract: {imgExtracts.Count} image{(imgExtracts.Count == 1 ? "" : "s")} from ";
+        foreach(var pi in pageIndices)
```
Comment (alexis-) [inline PDF/Viewer/IPDFViewer.SuperMemo.cs]: Can you use `string.Join(...)` instead ? It will be much cleaner and remove the need for `TrimEnd`

**Comment by bjsi** [comment]
OK. I cleaned up the code as you suggested. I also found out how to prepend the title to the extract title in the KT window. I think it looks pretty clean now.

image

image

**Comment by alexis-** [comment]
I'm adding this manually, it's easier than fixing the conflicts. I'm adding you to the commit with the Co-author tag to credit your work.

Thanks!

**Pull request closed without merging**